### PR TITLE
Eh 827 osaamisen hankkimistapa title

### PR DESCRIPTION
**Pull request by vesaruu**
Added new title for osaamisenHankkimistapa when osaamisenHankkimistapa type is now workplace. Moved muut oppimisympäristöt under learning period. Previously was rendered after all learningPeriods.

### CODE REVIEW

File: shared/components/StudyInfo/LearningPeriod.tsx [hunk 1]
```diff
@@ -46,41 +47,28 @@ export class LearningPeriod extends React.Component<LearningPeriodProps> {
...
       workplaceSelite,
       ajanjaksonTarkenne,
-      jarjestajanEdustaja
+      jarjestajanEdustaja,
+      muutOppimisymparistot
     } = learningPeriod
 
     const vastuullinenTyopaikkaOhjaaja =
-      tyopaikallaJarjestettavaKoulutus?.vastuullinenTyopaikkaOhjaaja
+      tyopaikallaJarjestettavaKoulutus.vastuullinenTyopaikkaOhjaaja
```
Comment (J-Kallunki) [inline shared/components/StudyInfo/LearningPeriod.tsx]: const {vastuullinenTyopaikkaOhjaaja, keskeisetTyotehtavat} = tyopaikallaJarjestettavaKoulutus

File: shared/components/StudyInfo/LearningPeriod.tsx [hunk 2]
```diff
@@ -100,7 +88,8 @@ export class LearningPeriod extends React.Component<LearningPeriodProps> {
                     />
                   </TH>
                   <TD>
-                    {vastuullinenTyopaikkaOhjaaja.nimi}, {selite}
+                    {vastuullinenTyopaikkaOhjaaja.nimi}
+                    {!!selite ? `, ${selite}` : ""}
```
Comment (J-Kallunki) [inline shared/components/StudyInfo/LearningPeriod.tsx]: {!!selite && `, ${selite}`} will also do fine

File: shared/components/StudyInfo/LearningPeriod.tsx [hunk 3]
```diff
@@ -149,6 +138,24 @@ export class LearningPeriod extends React.Component<LearningPeriodProps> {
             </HMediaQuery.MaxWidth>
           </React.Fragment>
         )}
+
+        {muutOppimisymparistot.map((environment, i) => {
+          const nimi =
```
Comment (J-Kallunki) [inline shared/components/StudyInfo/LearningPeriod.tsx]: const {nimi = ""} = environment.oppimisymparisto

File: shared/models/OsaamisenHankkimistapa.ts [hunk 4]
```diff
@@ -30,27 +30,18 @@ export const OsaamisenHankkimistapa = types
...
-          ? `${self.selite}, ${self.tyopaikallaJarjestettavaKoulutus.tyopaikanYTunnus}`
-          : self.selite
+          !!self.tyopaikallaJarjestettavaKoulutus.tyopaikanNimi
+          ? `${self.tyopaikallaJarjestettavaKoulutus.tyopaikanNimi}, ${self.tyopaikallaJarjestettavaKoulutus.tyopaikanYTunnus}`
+          : ""
       },
-      get tyyppi():
-        | OsaamisenHankkimistapaType.Workplace
-        | OsaamisenHankkimistapaType.Other {
+      get tyyppi(): OsaamisenHankkimistapaType {
```
Comment (J-Kallunki) [inline shared/models/OsaamisenHankkimistapa.ts]: TS will also handle tis type dynamically

File: shared/components/StudyInfo/LearningPeriod.tsx (hunk 3 above)
Comment (vesaruu) [reply shared/components/StudyInfo/LearningPeriod.tsx]: oppimisymparisto for muut environment is mandatory so optional checks could be removed

File: shared/models/OsaamisenHankkimistapa.ts (hunk 4 above)
Comment (vesaruu) [reply shared/models/OsaamisenHankkimistapa.ts]: true

File: shared/components/StudyInfo/LearningPeriod.tsx (hunk 1 above)
Comment (vesaruu) [reply shared/components/StudyInfo/LearningPeriod.tsx]: yes, that's cleaner

File: shared/components/StudyInfo/LearningPeriod.tsx (hunk 2 above)
Comment (vesaruu) [reply shared/components/StudyInfo/LearningPeriod.tsx]: I'll fix the underlying problem in a upcoming pr.

**Pull request merged**